test/test_cuda.py

```diff
@@ -2075,6 +2075,31 @@ def _worker(t):
             for t in range(num_threads):
                 self.assertEqual(results[t].sum().item(), size * size)
 
+    def testCudaRecordStreamWithWrongDevice(self):
+        t = torch.randn(5, 5)
+        with self.assertRaisesRegex(RuntimeError, "Tensor is not allocated in CUDA"):
+            t.record_stream(torch.cuda.current_stream())
+        # Copy to CUDA
+        ct = t.to("cuda")
+        ct.record_stream(torch.cuda.current_stream())
+
+    @classmethod
+    def _recordStreamFromADifferentProcess(cls, t):
+        inst = cls()
+        # This should throw exception as recrod_stream in a different process will not succeed
+        with inst.assertRaisesRegex(RuntimeError, "Tensor is not allocated by the CUDACachingAllocator instance in current context"):
+            t.record_stream(torch.cuda.current_stream())
+
+    def testCudaRecordStreamWithinWrongProcess(self):
+        SIZE = 5
+        t = torch.cuda.FloatTensor(SIZE)
+
+        ctx = mp.get_context('spawn')
+        p = ctx.Process(target=TestCuda._recordStreamFromADifferentProcess, args=(t,))
+        p.start()
+        p.join()
+        self.assertEqual(t.size()[0], SIZE)
+
```

Comment:
@ezyang @gchanan 

Is there a way to call `cudaMalloc` from Python so that we can test CUDA blocks not allocated by `CUDACachingAllocator`?